tern/classes/oci_image.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+import os
+import json
+import subprocess  # nosec
+from tern.utils import rootfs
+from tern.utils import general
+from tern.classes.image import Image
+from tern.utils.constants import manifest_file
+from tern.classes.image_layer import ImageLayer
+
+
+class OCIImage(Image):
+    '''A representation of an image created using OCI format
+    See image.py for super class's attributes
+    OCI Image specific attributes:
+        repotag: the repotag associated with this image
+        history: a list of commands used to create the filesystem layers
+        to_dict: return a dict representation of the object
+    '''
+
+    def __init__(self, repotag=None, image_id=None, image_name=None):
+        super().__init__(image_id)
+        self.__repotag = repotag
+        self.__history = None
+        self._name = os.path.basename(image_name)
+
+    @property
+    def repotag(self):
+        return self.__repotag
+
+    @property
+    def history(self):
+        return self.__history
+
+    def to_dict(self, template=None):
+        '''Return a dictionary representation of the OCI image'''
+        # this should take care of 'origins' and 'layers'
+        di_dict = super().to_dict(template)
+        return di_dict
+
+    def get_image_option(self):
+        '''Check to see which value was used to init the image object
+        Return the value that was used. If neither one was used raise
+        NameError. If both were used return the id'''
+        if self.repotag is not None and self.image_id is not None:
+            return self.image_id
+        if self.repotag is not None:
+            return self.repotag
+        if self.image_id is not None:
+            return self.image_id
+        raise NameError("Image object initialized with no repotag or ID")
+
+    def get_image_manifest(self):
+        '''Assuming that there is a temp folder with a manifest.json of
+        an image inside, get a dict of the manifest.json file'''
+        temp_path = rootfs.get_working_dir()
+        with general.pushd(temp_path):
+            with open(manifest_file) as f:
+                json_obj = json.loads(f.read())
+        return json_obj
```

Comment:
OCI images do not have a `manifest.json` file. They will have a `index.json` file instead. That's OK. We only need to know where the layers are located and how to unpack them.

Reply:
@nishakm I think we have both the files inside OCI image layout directory. `index.json` points to `manifest.json` and `manifest.json` points to `config` and `layers` files. That is why I kept docker image class code as it is also based on `manifest.json` logic. Only OCI image `repotag` information I am fetching from `index.json`. For example, I am attaching a screenshot of `busybox` OCI image layout here,

<img width="1910" alt="Screenshot 2020-06-26 at 11 33 05 PM" src="https://user-images.githubusercontent.com/16931639/85887528-9f965c00-b805-11ea-8f1c-8cb05ae0462b.png">

Reply:
@nishakm yes, you are right the implementation assumes that ` there is a temp folder with manifest.json existing at the top level.`. Actually I have already written the whole functionality which I shared through my previous PR. In that PR, I proposed a `helper` utility function to copy all the blobs such as `index.json / manifest.json / config & layers, etc` inside tern's `temp` folder before calling `load_image()`.  The intention was to create minimal changes to analyze OCI image and making the whole process similar to docker image analysis. Currently, with this PR I have introduced `oci_image.py` as you suggested.

Reply:
yes, I got this layout after executing the command you shared  `skopeo copy docker://docker.io/library/busybox oci:busybox`

<img width="1579" alt="Screenshot 2020-06-30 at 8 54 05 PM" src="https://user-images.githubusercontent.com/16931639/86145014-0b353d80-bb14-11ea-9d39-2cf3d8506c29.png">

Reply:
> @nishakm yes, you are right the implementation assumes that ` there is a temp folder with manifest.json existing at the top level.`. Actually I have already written the whole functionality which I shared through my previous PR. In that PR, I proposed a `helper` utility function to copy all the blobs such as `index.json / manifest.json / config & layers, etc` inside tern's `temp` folder before calling `load_image()`. The intention was to create minimal changes to analyze OCI image and making the whole process similar to docker image analysis. Currently, with this PR I have only introduced `oci_image.py` as you suggested.

For one, `index.json` is your way to finding where the config and layers are. I noticed that you read this file every time you want to load the config and layers. Rather than do that, you could read `index.json` and from there read blobs/<digest type>/<digest file>. Of course you could implement all of this functionality within `load_image()` if you wanted, but it leads to better readability if you had helper functions that would do the parsing bit by bit. These are very simple manifests. They can get super complicated in the future.

Reply:
@nishakm, as of now the idea is to read `index.json` only once when to copy the files inside `temp` folder as I mentioned earlier. After this, I am loading `manifest` inside `load_image` and getting all the information from there. 
Another, I have gone through the example you shared and if I understand it correctly then I think the `architecture` information is the key here to load the specific manifest. If it is correct then how should we get this information?